src/rgw/rgw_amqp.h

```diff
@@ -14,26 +14,48 @@ struct connection_t;
 typedef std::function<void(int)> reply_callback_t;
 
 // connect to an amqp endpoint
-const connection_t& connect(const std::string& url, const std::string& exchange);
+connection_t& connect(const std::string& url, const std::string& exchange);
 
 // publish a message over a connection that was already created
-int publish(const connection_t& conn,
+int publish(connection_t& conn,
     const std::string& topic,
     const std::string& message);
 
 // publish a message over a connection that was already created
 // and pass a callback that will be invoked (async) when broker confirms
 // receiving the message
-int publish_with_confirm(const connection_t& conn, 
+int publish_with_confirm(connection_t& conn, 
     const std::string& topic,
     const std::string& message,
     reply_callback_t cb);
 
 // convert the integer status returned from the "publish" function to a string
 std::string status_to_string(int s);
 
-// return the number of open connections
-unsigned get_connection_number();
+// number of connections
+size_t get_connection_number();
```

Comment:
"connection_number" sounds like an identifier for a unique connection, maybe "connection_count?"